ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/ConfigCommitter.scala

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils.committer
+
+import com.daml.ledger.participant.state.kvutils.Conversions._
+import com.daml.ledger.participant.state.kvutils.DamlKvutils._
+import com.daml.ledger.participant.state.kvutils.committing.Common.getCurrentConfiguration
+import com.daml.ledger.participant.state.v1.Configuration
+
+private[kvutils] object ConfigCommitter {
+  case class Result(
+      builder: DamlConfigurationEntry.Builder,
+      currentConfig: (Option[DamlConfigurationEntry], Configuration))
+}
+
+private[kvutils] case class ConfigCommitter(
+    defaultConfig: Configuration
+) extends Committer[DamlConfigurationSubmission, ConfigCommitter.Result] {
+
+  private object Metrics {
+    // kvutils.ConfigCommitter.*
+    val accepts = metricsRegistry.counter("accepts")
+    val rejections = metricsRegistry.counter("rejections")
+  }
+
+  private def rejectionTraceLog(
+      msg: String,
+      configurationEntry: DamlConfigurationEntry.Builder): Unit =
+    logger.trace(
+      s"Configuration rejected, $msg, correlationId=${configurationEntry.getSubmissionId}")
+
+  private val checkTtl: Step = (ctx, result) => {
```

Comment:
consider already in this PR pulling the checkTtl and authorizeSubmission to be built into Committer (we'll of course need to specify how to build the timeout/participant not authorized results).

Reply:
Haven't done that one. The commonalities are still too hard to spot through the thicket of diverging abstractions.